activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -1027,6 +1027,9 @@ def type_to_sql(type, limit = nil, precision = nil, scale = nil) #:nodoc:
 
           column_type_sql
         else
+          unless limit.nil? && precision.nil? && scale.nil?
+            raise(ActiveRecordError, "You've specified :limit, :precision, or :scale for type \"#{type}\" which does not support it")
```

Comment:
The other error messages in this file are...I can't think of a good description, less personal? 

For example we have one that says "No indexes found on #{table_name} with the options provided". I think it's better to have an error that says `\"#{type}\" is not a supported :limit, :precision, or :scale".